src/App.tsx

```diff
@@ -1,14 +1,34 @@
 /* eslint-disable max-len */
-import React from 'react';
-import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import 'bulma/css/bulma.css';
+import React, { useEffect, useState } from 'react';
 
-import { TodoList } from './components/TodoList';
+import { getTodos } from './api';
+import { Loader } from './components/Loader';
 import { TodoFilter } from './components/TodoFilter';
+import { TodoList } from './components/TodoList';
 import { TodoModal } from './components/TodoModal';
-import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [updatedTodos, setUpdatedTodos] = useState<Todo[]>(todos);
```

Comment:
Initializing state with props can lead to bugs and unexpected behavior. It is better to initialize your state to a default value, like an empty array. In this case, you are initializing `updatedTodos` with `todos` which is a state variable itself. This can lead to a situation where `updatedTodos` does not reflect the current state of `todos` because `useState` does not automatically sync with changes in `todos`. You might want to consider using a different approach to keep `todos` and `updatedTodos` in sync. You can find more information about this issue [here](https://stackoverflow.com/a/50403930).